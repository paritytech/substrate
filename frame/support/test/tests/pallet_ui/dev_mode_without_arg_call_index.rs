#![cfg_attr(not(feature = "std"), no_std)]

pub use pallet::*;

#[frame_support::pallet]
pub mod pallet {
	use frame_support::pallet_prelude::*;
	use frame_system::pallet_prelude::OriginFor;

	// The struct on which we build all of our Pallet logic.
	#[pallet::pallet]
	pub struct Pallet<T>(_);

	// Your Pallet's configuration trait, representing custom external types and interfaces.
	#[pallet::config]
	pub trait Config: frame_system::Config {}

	// Your Pallet's callable functions.
	#[pallet::call]
	impl<T: Config> Pallet<T> {
		#[pallet::weight(0)]
		pub fn my_call(_origin: OriginFor<T>) -> DispatchResult {
			Ok(())
		}
	}

	// Your Pallet's internal functions.
	impl<T: Config> Pallet<T> {}
}

fn main() {}
