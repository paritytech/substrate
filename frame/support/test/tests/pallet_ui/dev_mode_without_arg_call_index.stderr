error: use of deprecated constant `pallet::warnings::ImplicitCallIndex_0::_w`:
               It is deprecated to use implicit call indices.
               Please instead ensure that all calls have a `pallet::call_index` attribute or put the pallet into `dev` mode.

               For more info see:
                   <https://github.com/paritytech/substrate/pull/12891>
                   <https://github.com/paritytech/substrate/pull/11381>
  --> tests/pallet_ui/dev_mode_without_arg_call_index.rs:22:10
   |
22 |         pub fn my_call(_origin: OriginFor<T>) -> DispatchResult {
   |                ^^^^^^^
   |
   = note: `-D deprecated` implied by `-D warnings`

error: use of deprecated constant `pallet::warnings::ConstantWeight_0::_w`:
               It is deprecated to use hard-coded constant as call weight.
               Please instead benchmark all calls or put the pallet into `dev` mode.

               For more info see:
                   <https://github.com/paritytech/substrate/pull/13798>
  --> tests/pallet_ui/dev_mode_without_arg_call_index.rs:21:20
   |
21 |         #[pallet::weight(0)]
   |                          ^
